src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -157,42 +157,184 @@ public Task<RestoreResult> ExecuteAsync()
             return ExecuteAsync(CancellationToken.None);
         }
 
-        public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
+        private void InitializeTelemetry(TelemetryActivity telemetry)
```

Comment:
Generally, the first method in a class tends to be a public method. 
Sometimes, the most important method is on top as well.

I also think it might minimize the number of changes we see, make the main part easier to review.

So my suggestion is move ExecuteAsync on top.